### PR TITLE
Bumping golang versions and adding 1.17

### PR DESCRIPTION
**Pull request by pokearu**
*Description of changes:*
Adding Go 1.17 to builder base and bumping other golang versions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: builder-base/install.sh [hunk 1]
```diff
@@ -50,7 +50,7 @@ yum install -y \
     unzip \
     wget
 
-GOLANG_VERSION="${GOLANG_VERSION:-1.16.7}"
+GOLANG_VERSION="${GOLANG_VERSION:-1.16.12}"
```
Comment (jaxesn) [inline builder-base/install.sh:53]: youll need to update the checksum files for this one

**Review by jaxesn — COMMENTED** [review]
When this is merged and a new image is pushed, we should probably trigger the two attribution periodic jobs which will also build everything and open PRs to update any changed checksums.  If we dont, then builds in build-tooling may start "randomly" failing due to checksum mismatches.

File: builder-base/install.sh (hunk 1 above)
Comment (pokearu) [reply builder-base/install.sh:53]: Got it thanks!

**Review by abhay-krishna — APPROVED** [review]
/lgtm

**Pull request merged**